src/utils/common.ts

```diff
@@ -265,9 +265,10 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
+  const gasMultiplier = 1.0; // Don't apply any multiplier to the gas price.
   const queries = [
     gasUnits ? Promise.resolve(BigNumber.from(gasUnits)) : voidSigner.estimateGas(unsignedTx),
-    _gasPrice ? Promise.resolve({ maxFeePerGas: _gasPrice }) : getGasPriceEstimate(provider, chainId, transport),
+    _gasPrice ? Promise.resolve({ maxFeePerGas: _gasPrice }) : getGasPriceEstimate(provider, chainId, gasMultiplier, transport),
```

Comment:
@nicholaspai I updated this to fix a merge conflcit after merging my PR, but I'm curious about why we don't want to apply any buffer here. This is where the API would otherwise want to apply it, if I've understood correctly.

Reply:
Oh you're right this is a mistake. I only applied it to the fresh query to the oracle but not to the input value if defined

Reply:
Is this what you were thinking? 4a0858a4b39738b494db70ae378e91275cc1800e

Reply:
@nicholaspai I'm not sure tbh. Do we want to apply the multiplier if the caller specifies the gasPrice? There is a risk that we revert to the recursive gas price multiplication that we had a while ago (i.e. the API sources a gas price including multiplier, then supplies it back to the gasPriceOracle and it gets multiplied _again_).

But I was also looking at line 268, where we have `const gasMultiplier = 1.0`. I think we want that to be user-supplied, rather than hardcoded in this file, right?

Reply:
You're right this is a mistake. We should allow caller to pass in gas price multiplier and NOT apply it if the gas price is passed in.